### PR TITLE
Added e2e tests for `page.tag.detached` webhook

### PR DESCRIPTION
**Pull request by Dark-Knight11**
refs: #15537

This adds an e2e test and test snapshot for the `page.tag.detached` webhook so we can prevent regressions and bugs in the future

Got some code for us? Awesome 🎊!

Please include a description of your change & check your PR against this list, thanks!

- [x] There's a clear use-case for this code change, explained below
- [x] Commit message has a short title & references relevant issues
- [x] The build will pass (run `yarn test:all` and `yarn lint`)

We appreciate your contribution!

Also, if you'd be interested in writing code like this for us more regularly, we're hiring:
https://careers.ghost.org

### CODE REVIEW

File: ghost/core/test/e2e-webhooks/pages.test.js [hunk 1]
```diff
@@ -56,13 +68,47 @@ const buildPageSnapshotWithTiers = ({
     };
 };
 
+const buildPageSnapshotWithTiersAndTags = ({
```
Comment (naz) [inline ghost/core/test/e2e-webhooks/pages.test.js:71]: Would it make sense to reuse the `buildPageSnapshotWithTiers` method here?

File: ghost/core/test/e2e-webhooks/pages.test.js [hunk 2]
```diff
@@ -12,9 +12,21 @@ const {
...
     anyContentVersion,
-    anyNumber
+    anyNumber,
+    anyString
 } = matchers;
 
+const tagSnapshot = {
+    created_at: anyISODateTime,
+    id: anyObjectId,
+    name: anyString,
```
Comment (naz) [inline ghost/core/test/e2e-webhooks/pages.test.js:22]: The tag's name, slug, visibility etc. are not changing between test runs? If they are not, would be best to have snapshot capture them as is - no need to include non-dynamic properties in the snapshot matcher object 👍

**Review by naz — CHANGES_REQUESTED** [review]
Hey-hey @Dark-Knight11 👋  These are great changes are I think we are 90% there. I've left couple of comments around areas that need improvement. Feel free to reach out if you need more guidance and thanks for the contribution!

**Comment by Dark-Knight11** [comment]
> Hey-hey @Dark-Knight11 👋 These are great changes are I think we are 90% there. I've left couple of comments around areas that need improvement. Feel free to reach out if you need more guidance and thanks for the contribution!

Hi, I have made the requested changes kindly look into it and lmk if anymore changes are needed from my side.
Thanks!

**Comment by naz** [comment]
The snapshots seem to be outdated on and are failing now. Can you please re-record them again @Dark-Knight11 🙏

**Review by naz — APPROVED** [review]
Great work on these changes @Dark-Knight11! Looking forward for more contributions 😊

**Comment by ErisDS** [comment]
Hey @Dark-Knight11 thanks again for your contributions! This has been merged and the `hacktoberfest-accepted` label added to make sure it counts.

Please keep them coming 🙏 we have so much we want to achieve beyond hacktoberfest 😬

**Pull request merged**